A utility function to read vtu solution files.

alarshi opened the pull request:
Currently, we are using `meshio` library to read the ASPECT solution files. This PR adds the functions to read the solution files in a separate directory because we will use it at several places in the modules.

alarshi committed:
A utility function to read vtu solution files.

naliboff commented on src/utilities.py:
```diff
@@ -0,0 +1,51 @@
+import pandas as pd
+
+def read_statistics(fname):
+    """ Read the statistics file output by CIG-ASPECT
```
Have we decided on a consistent docstrings format? I think using the Numpy one (see example [here](https://www.geeksforgeeks.org/python-docstrings/)) would be a good standard to follow.

naliboff replied:
Ah, you use that format on the function below. Can you apply it here as well?

naliboff commented on src/utilities.py:
```diff
@@ -0,0 +1,51 @@
+
+def read_vtu_file (file_name):
+    """
+    This function takes an input .vtu file_name (either structured or
```
```suggestion
    This function takes an input .vtu file name from a structured or
```

naliboff commented on src/utilities.py:
```diff
@@ -0,0 +1,51 @@
+def read_vtu_file (file_name):
+    """
+    This function takes an input .vtu file_name (either structured or
+    unstructured mesh) and returns the point coordinates and values in that file.  
```
```suggestion
    unstructured mesh, and returns the point coordinates and values in that file.  
```

naliboff reviewed (changes requested):
@alarshi - Thanks for contributing and this looks good other than a few minor formatting comments.

General questions:
1. Should we begin to do automated testing through github actions and add tests for each new function.
2. In the PR discussion, can you provide a demonstration of how the functions are used?

alarshi committed:
address comments from the PR.

alarshi commented:
@naliboff : Thank you for the review! I addressed your suggestions in the latest commit. 

1. That's a good idea but so far we only functions for data I/O and plotting. I am not sure how to create tests for such cases but we can discuss that in our meeting. 
2. If we were to plot temperature field in the model , we can do so by reading it through the utilities and then plotting in using matplotlib functions.

```
    mesh_nodes, sample_data = utilities.read_vtu_file(f)
    points_x = mesh_nodes[:, 0]
    points_y = mesh_nodes[:, 1]

    # Create interpolation grid 
    x_grid = np.linspace(min(points_x), max(points_x), len(points_x))
    y_grid = np.linspace(min(points_y), max(points_y), len(points_y))
    X, Y = np.meshgrid(x_grid, y_grid, indexing='ij')
    
    interp_data = griddata((points_x, points_y), sample_data['T'], (X, Y), method='linear')
```

The `read_statistics` function is taken from the contrib directory in ASPECT and will be used to plot any statistics generated by ASPECT.

naliboff reviewed (approved):
👍
